Add Island Invite System

Mohron opened the pull request:
- Added `/is invite` - `skyclaims.command.invite`
- Added `/is kick` - `skyclaims.command.kick`
- Added` /is leave` - `skyclaims.command.leave`
- Added` /is promote` - `skyclaims.command.promote`
- Added` /is demote` - `skyclaims.command.demote`
- Use of GriefPrevention's `/trust` & `/permissiontrust` on an island is handled by `/is invite`


Will replace #20

Mohron committed:
Add Island Invite System

- Add /is invite - skyclaims.command.invite
- Add /is kick - skyclaims.command.kick
- Add /is leave - skyclaims.command.leave
- Use of GriefPrevention's /trust & /permissiontrust on an island is handled by /is invite

Mohron commented on README.md:
```diff
@@ -17,24 +17,29 @@ This plugin is in Beta. Live support is available for the latest builds through
 - [X] Isolation of islands to their own Minecraft region file
-- [X] Allow spawn/tp on your island at a configurable location 
-- [X] Use GriefPrevention trust commands to work in teams
+- [X] Allow spawn/tp on your island at a configurable location
```
@subtank would you be able to translate these readme changes?

subtank replied:
yes.i am working on this.

subtank replied:
@Mohron working on chinese change log.

subtank committed:
Update README.zh-CN.md (#60)

* Update README.zh-CN.md

Mohron committed:
Remove readme comment

subtank commented on CHANGELOG.md:
```diff
@@ -1,14 +1,27 @@
+## Upcoming/Unreleased Features, Changes & Bugfixes
+**REQUIRED: SF build 2558+ & GP build 4.0.0.415+ OPTIONAL: Nucleus version 1.1.3+**
+- Added Void World Generation capabilities
+- Added Nether portal fix
```
???did you mean the nether portal did not work?

Mohron replied:
Don't worry about the Unreleased/Upcoming changes as none of this is set in stone. But what's happening is people are being sent to nether portals they can't access because it's on an island they don't have access to.

SubtypeZero replied:
Didn't realize you had changed this, will have to check it out soon

Mohron replied:
Those things aren't in this PR, they're stashed on my PC while I address issues with it.

SubtypeZero replied:
Oh,  duh. Let me know if you'd like for me to test changes to nether portals. I had some issues with those in the past that I'd like to get fixed.

Mohron replied:
I could probably sneak in half of the planned nether portal changes I want to make available in this build tbh.

subtank commented on CHANGELOG.md:
```diff
@@ -1,14 +1,27 @@
+    - `skyclaims.command.list.unlocked` - allows a player to list all unlocked islands
+    - `skyclaims.command.list.sort` - allows a player to use the list sorting features
+    - `skyclaims.admin.list` - allows a player to list all islands
+- Added _Misc/Teleport-on-Creation_ config to disable automatic teleportation after island creation
```
what's this means?

Mohron replied:
When an island is created either on-join or with `/is create`, the plugin automatically teleports the player after it is created. I added a config to disable that feature (by request).

subtank commented on CHANGELOG.md:
```diff
@@ -1,14 +1,27 @@
+- Added _Misc/Teleport-on-Creation_ config to disable automatic teleportation after island creation
+- `/is create` & `/is reset` now support generating a clickable list from available schematics
 - Fixed an IndexOutOfBoundsException with the entity limits feature
+- Added Void Island Control Schematics to prepackaged schematics
```
Void Island Control Schematics
is this a new schematics?

Mohron replied:
https://minecraft.curseforge.com/projects/void-island-control
^ These are now included by default just to help servers get started quicker.

subtank replied:
will.i try it now and translation later for best translate

subtank replied:
you can finish this pull request and update release.

Mohron replied:
It's a big feature addition. Needs further testing before it's ready for release!

subtank replied:
will...can you send me an testing release to my mailbox?

Mohron replied:
You can get it on our Discord... https://discord.gg/EkVQycV

Mohron committed:
Add is promote & is demote commands

- Fix bug where invite can't be sent because another exists but really doesn't
- Fix bug where the player kicking another player gets the message intended for the player kicked

Mohron committed:
Bugfixes

Mohron committed:
Formating changes

Mohron committed:
More Formatting

Mohron committed:
Tooltip changes

Mohron commented on src/main/java/net/mohron/skyclaims/config/type/MiscConfig.java:
```diff
@@ -51,4 +56,13 @@ public boolean isTeleportOnCreate() {
+            return new SimpleDateFormat(dateFormat);
+        } catch (IllegalArgumentException e) {
+            SkyClaims.getInstance().getLogger().info("Invalid Date Format: {}", dateFormat);
+            return new SimpleDateFormat("dd MMM yy  hh:mm a");
```
Note to self: fix this...

Mohron commented on src/main/java/net/mohron/skyclaims/config/type/WorldConfig.java:
```diff
@@ -39,7 +39,7 @@
     private int spawnRegions = 1;
     @Setting(value = "List-Schematics")
-    private boolean listSchematics= true;
+    private boolean listSchematics = true;
```
Move to Misc

Mohron committed:
Fix some configs
